src/manage_nsfs/upgrade.js

```diff
@@ -0,0 +1,79 @@
+/* Copyright (C) 2024 NooBaa */
+'use strict';
+
+const dbg = require('../util/debug_module')(__filename);
+const { ManageCLIError } = require('./manage_nsfs_cli_errors');
+const { UPGRADE_ACTIONS } = require('./manage_nsfs_constants');
+const { ManageCLIResponse } = require('../manage_nsfs/manage_nsfs_cli_responses');
+const { throw_cli_error, write_stdout_response } = require('./manage_nsfs_cli_utils');
+
+/**
+ * manage_upgrade_operations handles cli upgrade operations
+ * @param {string} action 
+ * @returns {Promise<Void>}
+ */
+async function manage_upgrade_operations(action, config_fs) {
+    switch (action) {
+        case UPGRADE_ACTIONS.START:
+            await exec_config_dir_upgrade();
+            break;
+        case UPGRADE_ACTIONS.STATUS:
+            await get_upgrade_status(config_fs);
+            break;
+        case UPGRADE_ACTIONS.HISTORY:
+            await get_upgrade_history(config_fs);
+            break;
+        default:
+            throw_cli_error(ManageCLIError.InvalidAction);
```

Comment:
can we use upgrade specific error here, I know we do the validation before reaching here but since we have more specific validation I think we can use that 
`ManageCLIError.InvalidUpgradeAction`